java/cdk/src/main/java/sleeper/cdk/SleeperCdkApp.java

```diff
@@ -235,13 +238,22 @@ public void create() {
         }
 
         // Stack to execute queries
-        if (optionalStacks.contains(QueryStack.class.getSimpleName())) {
-            new QueryStack(this,
+        if (QUERY_STACK_NAMES.stream().anyMatch(optionalStacks::contains)) {
+            QueryQueueStack queryQueueStack = new QueryQueueStack(this, "QueryQueue", instanceProperties);
+            queryStack = new QueryStack(this,
                     "Query",
                     instanceProperties, jars,
-                    coreStacks);
+                    coreStacks, queryQueueStack);
+            // Stack to execute queries using the web socket API
+            if (optionalStacks.contains(WebSocketQueryStack.class.getSimpleName())) {
+                WebSocketQueryStack webSocketQueryStack = new WebSocketQueryStack(this,
+                        "WebSocketQuery",
+                        instanceProperties, jars,
+                        coreStacks, queryQueueStack);
+                webSocketQueryStack.grantStageApiAccess(queryStack.getQueryExecutorLambda());
+                webSocketQueryStack.grantStageApiAccess(queryStack.getLeafPartitionQueryLambda());
```

Comment:
Please could the QueryStack be passed into the WebSocketQueryStack so that the permissions can be configured in one place?